cnd/src/db/save.rs

```diff
@@ -60,27 +62,71 @@ impl From<Swap> for InsertableSwap {
 #[table_name = "rfc003_bitcoin_ethereum_bitcoin_ether_request_messages"]
 struct InsertableBitcoinEthereumBitcoinEtherRequestMessage {
     swap_id: Text<SwapId>,
-    bitcoin_network: Text<bitcoin::Network>,
+    bitcoin_network: Text<BitcoinNetwork>,
     ethereum_chain_id: U32,
     bitcoin_amount: Text<Satoshis>,
     ether_amount: Text<Ether>,
     hash_function: Text<HashFunction>,
-    bitcoin_refund_identity: Text<bitcoin::PublicKey>,
+    bitcoin_refund_identity: Text<::bitcoin::PublicKey>,
     ethereum_redeem_identity: Text<EthereumAddress>,
     bitcoin_expiry: U32,
     ethereum_expiry: U32,
     secret_hash: Text<SecretHash>,
 }
 
 #[async_trait]
-impl Save<Request<Bitcoin, Ethereum, asset::Bitcoin, asset::Ether>> for Sqlite {
+impl Save<Request<ledger::bitcoin::Regtest, Ethereum, asset::Bitcoin, asset::Ether>> for Sqlite {
```

Comment:
We could use a macro here to fore the three types of network, the imlp blocks only seem to differ by a single type.

Reply:
At this stage trying to make it work. @thomaseizinger suggested to leave the code optimisation for a later day.